cylc/flow/cfgspec/workflow.py

```diff
@@ -1362,24 +1394,92 @@ def get_script_common_text(this: str, example: Optional[str] = None):
-                Conf('expired handlers', VDR.V_STRING_LIST, None)
-                Conf('late offset', VDR.V_INTERVAL, None)
-                Conf('late handlers', VDR.V_STRING_LIST, None)
-                Conf('submitted handlers', VDR.V_STRING_LIST, None)
-                Conf('started handlers', VDR.V_STRING_LIST, None)
-                Conf('succeeded handlers', VDR.V_STRING_LIST, None)
-                Conf('failed handlers', VDR.V_STRING_LIST, None)
-                Conf('submission failed handlers', VDR.V_STRING_LIST, None)
-                Conf('warning handlers', VDR.V_STRING_LIST, None)
-                Conf('critical handlers', VDR.V_STRING_LIST, None)
-                Conf('retry handlers', VDR.V_STRING_LIST, None)
-                Conf('submission retry handlers', VDR.V_STRING_LIST, None)
-                Conf('execution timeout handlers', VDR.V_STRING_LIST, None)
-                Conf('submission timeout handlers', VDR.V_STRING_LIST, None)
-                Conf('custom handlers', VDR.V_STRING_LIST, None)
+                Conf('expired handlers', VDR.V_STRING_LIST, None, desc='''
+                    Handlers to invoke if this task has expired.
+
+                    .. caution::
+
+                       Changes to the scheduling algorithm in Cylc 8 mean
+                       this event will not be triggered until the expired task
+                       is ready to run.  Earlier expired-task detection will be
+                       implemented in a future Cylc release.
+                ''')
+                Conf('late offset', VDR.V_INTERVAL, None, desc='''
+                    Handlers to invoke if the task starts running behind
+                    schedule.
+
+                    Offset from cycle point, in real time, at which this task
+                    is considered to be "running late" (i.e. the time by which
+                    it would normally have started running).
+
+                    .. caution::
+
+                       Changes to the scheduling algorithm in Cylc 8 mean
+                       this event will not be triggered until the late task
+                       is ready to run.  Earlier late-task detection will be
+                       implemented in a future Cylc release.
+
+                    .. seealso::
+
+                       :cylc:conf:`flow.cylc[runtime][<namespace>][events]
+                       late handlers`.
+                ''')
+                Conf('late handlers', VDR.V_STRING_LIST, None, desc='''
+                    Handlers to be invoked if this task is late.
+
+                    .. caution::
+
+                       Due to changes to the Cylc 8 scheduling algorithm
+                       this event is unlikely to occur until the task is about
+                       to submit anyway.
+                ''')
+                Conf('submitted handlers', VDR.V_STRING_LIST, None, desc='''
+                    Handlers to invoke if this task is submitted.
```

Comment:
Perhaps replace "invoke" with "run" for all items?

Reply:
Done - will push